common/components/UpdateModal.tsx

```diff
@@ -0,0 +1,141 @@
+  };
+
+  public componentDidMount() {
+    addListener('UPDATE:update-downloaded', () => {
+      sendEvent('UPDATE:quit-and-install');
+    });
+    addListener('UPDATE:download-progress', downloadProgress => {
+      this.setState({ downloadProgress });
+    });
+    addListener('UPDATE:error', err => {
+      console.error('Update failed:', err);
+      this.setState({ isDownloading: false });
+      this.props.showNotification(
+        'danger',
+        <span>
+          Update could not be downloaded, please visit{' '}
+          <NewTabLink href="https://github.com/MyEtherWallet/MyEtherWallet/releases">
+            our github
+          </NewTabLink>{' '}
+          to download the latest release
+        </span>,
+        Infinity
+      );
+    });
+  }
+
+  public render() {
+    const { isOpen, updateInfo, handleClose } = this.props;
+    const { isDownloading, downloadProgress } = this.state;
+    const buttons: IButton[] | undefined = downloadProgress
+      ? undefined
+      : [
+          {
+            text: <span>{isDownloading && <Spinner />} Download Update</span>,
+            type: 'primary',
+            onClick: this.downloadUpdate,
+            disabled: isDownloading
+          },
+          {
+            text: 'Close',
+            type: 'default',
+            onClick: handleClose
+          }
+        ];
+
+    return (
+      <Modal
+        isOpen={isOpen}
+        title={`Update Information`}
+        handleClose={handleClose}
+        buttons={buttons}
+      >
+        <div className="UpdateModal">
+          {downloadProgress ? (
+            <div className="UpdateModal-downloader">
+              <h3 className="UpdateModal-downloader-title">Downloading...</h3>
+              <div className="UpdateModal-downloader-bar">
+                <div
+                  className="UpdateModal-downloader-bar-inner"
+                  style={{
+                    width: `${downloadProgress.percent}%`
+                  }}
+                />
+              </div>
+              <div className="UpdateModal-downloader-info">
+                <span className="UpdateModal-downloader-info-bit">
+                  Downloaded {downloadProgress.percent.toFixed(1)}%
+                </span>
+                <span className="UpdateModal-downloader-info-bit">
+                  {bytesToHuman(downloadProgress.bytesPerSecond)}/s
+                </span>
+                <span className="UpdateModal-downloader-info-bit">
+                  Total Size {bytesToHuman(downloadProgress.total)}
+                </span>
+              </div>
+            </div>
+          ) : (
+            <div>
+              <h1 className="UpdateModal-title">{updateInfo.releaseName}</h1>
+              <div className="UpdateModal-date">{moment(updateInfo.releaseDate).format('LL')}</div>
+              <div
+                className="UpdateModal-content"
+                dangerouslySetInnerHTML={{
+                  __html: updateInfo.releaseNotes
```

Comment:
Does this have any security implications (XSS)? Or will we always have full control over release notes

Reply:
Release notes are taken directly from Github releases. So unless Github is compromised or their flavor of markdown introduces the ability for XSS attacks, it should be fine. I wish they sent the markdown instead so that we could markdown process it, but unfortunately in comes in as HTML.